lib/services/oauth2_service.ex

```diff
@@ -50,7 +52,14 @@ defmodule EaRestaurantDataLoader.Oauth2Service do
   end
 
   defp validate_client_credentials(client, client_secret, entity_status) do
-    case Bcrypt.verify_pass(client_secret, client.client_secret) do
+    encoding_struct =
```

Comment:
I would love to see this simplified, something as simple as:

```
password = ApplicationUtil.build_password_type_from_env(password, encoded_password)

... PasswordEncoder.validate_password(password ) ...
```

Reply:
Done.